docs/integration/kafka.md

```diff
@@ -1,8 +1,181 @@
+
+For example, on CentOS, you need to add Sematext Linux packages and install them with the following command:
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that, set up the Kafka Monitoring Agent on your Kafka broker by running a command like this:
+```bash
+sudo bash /opt/spm/bin/setup-sematext  \
+    --monitoring-token <your-monitoring-token-goes-here>   \
+    --app-type kafka  \
+    --app-subtype kafka-broker  \
+    --agent-type javaagent  \
+    --infra-token <your-infra-token-goes-here>
+```
+
+Keep in mind that your need to provide the Monitoring token and Infra token. They are both provided in the [installation instructions](https://apps.sematext.com/ui/howto/Kafka/overview) for your Kafka App. 
+
+The last thing that needs to be done is adjusting the ```$KAFKA_HOME/bin/kafka-server-start.sh``` file and add the following section to the ```KAFKA_JMX_OPTS```:
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-broker:default
+```
+
+**You need to restart your Kafka broker after the changes above.**
+
+To see the complete picture of Kafka performance install the monitoring agent on each of your Kafka producers and consumers. Here is how you can do that.
+
+### Monitoring Producers
+
+To have the full visibility into the entire Kafka pipeline it's crucial to monitor your Kafka producers as well. If you're using Java or Scala as the language of choice for the producers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka producer by running the following command (e.g. for CentOS):
+
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that, run the following command to set up Kafka producer monitoring:
+
+```bash
+sudo bash /opt/spm/bin/setup-sematext  \
+    --monitoring-token <your-monitoring-token-goes-here>  \
+    --app-type kafka  \
+    --app-subtype kafka-producer  \
+    --agent-type javaagent  \
+    --infra-token <your-infra-token-goes-here>
+```
+
+Once that is done you need to add the following options to the JVM start-up properties:
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-producer:default
+```
+
+**You need to restart your Kafka producer after the changes above.**
+
+### Monitoring Consumers
+
+Monitoring your consumers is crucial to have visibility into topic lag, which can help you quickly identify issues with your pipeline. If you're using Java or Scala as the language of choice for the consumers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka consumer by running the following command (e.g. for CentOS):
```

Comment:
topic lag or consumer lag?

Reply:
Yes, consumer lag, not topic lag, you are right. The first version was good, this is not, adjusted. Good catch, thanks!